project/plugins/bitbucket/__init__.py

```diff
@@ -15,56 +14,148 @@ def getSecret(configMap):
           if config_key['plugin']+':'+config_key['tag'] == 'bitbucket:prod':
                return config_key['secret']
 
+def requestCredentials(configMap, data):
+    try:
+        responce = requests.post("https://bitbucket.org/site/oauth2/access_token",
+                                   auth=(getKey(configMap), getSecret(configMap)), data=data)
+    except requests.ConnectionError:
+        print("Could not connect to Bitbucket")
 
-def inviteUser(email,configMap,allPermissions, plugin_tag, name):
+    return responce
 
-     #Get Authorization token
-     data = {'grant_type': 'client_credentials'}
-     credential = requests.post("https://bitbucket.org/site/oauth2/access_token", auth=(getKey(configMap),getSecret(configMap)), data=data)
-     my_json = credential.content.decode('utf8')
-     data = json.loads(my_json)
-     access_token=data.get('access_token')
+def userExists(allMembers, userName, cli_groups):
 
-     cli_groups = []
-     for permission in allPermissions:
-         thisPermissions = ast.literal_eval(permission)
-         if thisPermissions['plugin'] == plugin_tag:
-             del thisPermissions['plugin']
-             cli_groups = list(thisPermissions.values())
-             break
+    for group in allMembers:
+        if group['name'].lower() in cli_groups:
+            for user in group['members']:
+                if userName == user['username']:
+                    return True
+    return False
 
-     if len(cli_groups) == 0:
-         cli_groups = getGroups(configMap,plugin_tag)
 
-     for group in cli_groups:
-         invGroup = requests.put(
-             "https://api.bitbucket.org/1.0/users/"+configMap['global']['organization']+"/invitations/"
-             + email + "/"+configMap['global']['organization']+"/"+group+ "?access_token=" + access_token)
+def inviteUser(email,configMap,allPermissions, plugin_tag, name):
 
+     done = False
+     userName = email.split('@', 1)[0]
+     #Get Authorization token
+     data = {'grant_type': 'client_credentials'}
 
-     log = 'BitBucket: Email invite sent from Bitbucket.\n'
-     instruction = inviteMessage(configMap,plugin_tag)
-     return getJsonResponse('Bitbucket', email, log, instruction)
+     credentials = requestCredentials(configMap, data)
+     log = errorCheck(credentials, plugin_tag, userName)
+
+     if log:
+         print(log)
+         instruction = log
+     else:
+         my_json = credentials.content.decode('utf8')
+         data = json.loads(my_json)
+         access_token=data.get('access_token')
+
+         cli_groups = []
+         for permission in allPermissions:
+             thisPermissions = ast.literal_eval(permission)
```

Comment:
What does ast.literal_eval do?

Reply:
it is used for safely evaluating strings containing Python expressions from untrusted sources